pennylane_lightning/lightning_gpu/lightning_gpu.py

```diff
@@ -74,135 +74,25 @@
-        "C(SingleExcitationPlus)",
-        "C(DoubleExcitation)",
-        "C(DoubleExcitationMinus)",
-        "C(DoubleExcitationPlus)",
-        "C(MultiRZ)",
-        "C(GlobalPhase)",
-        "Hadamard",
-        "S",
-        "Adjoint(S)",
-        "T",
-        "Adjoint(T)",
-        "SX",
-        "Adjoint(SX)",
-        "CNOT",
-        "SWAP",
-        "ISWAP",
-        "PSWAP",
-        "Adjoint(ISWAP)",
-        "SISWAP",
-        "Adjoint(SISWAP)",
-        "SQISW",
-        "CSWAP",
-        "Toffoli",
-        "CY",
-        "CZ",
-        "PhaseShift",
-        "ControlledPhaseShift",
-        "RX",
-        "RY",
-        "RZ",
-        "Rot",
-        "CRX",
-        "CRY",
-        "CRZ",
-        "CRot",
-        "IsingXX",
-        "IsingYY",
-        "IsingZZ",
-        "IsingXY",
-        "SingleExcitation",
-        "SingleExcitationPlus",
-        "SingleExcitationMinus",
-        "DoubleExcitation",
-        "DoubleExcitationPlus",
-        "DoubleExcitationMinus",
-        "Adjoint(MultiRZ)",
-        "Adjoint(GlobalPhase)",
-        "Adjoint(PhaseShift)",
-        "Adjoint(ControlledPhaseShift)",
-        "Adjoint(RX)",
-        "Adjoint(RY)",
-        "Adjoint(RZ)",
-        "Adjoint(CRX)",
-        "Adjoint(CRY)",
-        "Adjoint(CRZ)",
-        "Adjoint(IsingXX)",
-        "Adjoint(IsingYY)",
-        "Adjoint(IsingZZ)",
-        "Adjoint(IsingXY)",
-        "Adjoint(SingleExcitation)",
-        "Adjoint(SingleExcitationPlus)",
-        "Adjoint(SingleExcitationMinus)",
-        "Adjoint(DoubleExcitation)",
-        "Adjoint(DoubleExcitationPlus)",
-        "Adjoint(DoubleExcitationMinus)",
-        "QubitCarry",
-        "QubitSum",
-        "OrbitalRotation",
-        "ECR",
-        "BlockEncode",
-        "C(BlockEncode)",
-    }
-)
-# End the set of supported operations.
-
-# The set of supported observables.
-_observables = frozenset(
-    {
-        "PauliX",
-        "PauliY",
-        "PauliZ",
-        "Hadamard",
-        "SparseHamiltonian",
-        "LinearCombination",
-        "Hermitian",
-        "Identity",
-        "Projector",
-        "Sum",
-        "Prod",
-        "SProd",
-        "Exp",
-    }
-)
+_to_matrix_ops = {
+    "BlockEncode": OperatorProperties(controllable=True),
+    "ControlledQubitUnitary": OperatorProperties(),
+    "ECR": OperatorProperties(),
+    "SX": OperatorProperties(),
+    "ISWAP": OperatorProperties(),
+    "PSWAP": OperatorProperties(),
+    "SISWAP": OperatorProperties(),
+    "SQISW": OperatorProperties(),
+    "OrbitalRotation": OperatorProperties(),
+    "QubitCarry": OperatorProperties(),
+    "QubitSum": OperatorProperties(),
+    "DiagonalQubitUnitary": OperatorProperties(),
+}
 
 
 def stopping_condition(op: Operator) -> bool:
     """A function that determines whether or not an operation is supported by ``lightning.gpu``."""
-    return op.name in _operations
+    return LightningGPU.capabilities.supports_operation(op.name)
```

Comment:
What about caching these supported ops and obs methods as protected (private) callables to avoid Python namespace resolutions (i.e., importing `supports_op` `supports_obs` methods from `LightningGPU.capabilities` in multiple places)?

Reply:
What do you mean? I'm not very familiar with namespace resolution.

Reply:
Thanks!